aspnetcore/blazor/tutorials/movie-database-app/part-2.md

```diff
@@ -0,0 +1,533 @@
+* `-m`: The name of the model.
+* `-outDir`: The output directory for the generated components. A folder is created from the model name in the output directory to hold the components (for example, `MoviePages` in this case).
+
+:::zone-end
+
+The `appsettings.json` file is updated with the connection string used to connect to a local database. In the following example, the `{CONNECTION STRING}` is the connection string automatically generated by the scaffolder:
+
+```json
+"ConnectionStrings": {
+  "BlazorWebAppMoviesContext": "{CONNECTION STRING}"
+}
+```
+
+## Files created and updated by scaffolding
+
+The scaffolding process creates the following component files and movie database context class file:
+
+* `Components/Pages/MoviePages`
+  * `Create.razor`: Creates new movie entities.
+  * `Delete.razor`: Deletes a movie entity.
+  * `Details.razor`: Shows movie entity details.
+  * `Edit.razor`: Updates a movie entity.
+  * `Index.razor`: Lists movie entities (records) in the database.
+* `Data/BlazorWebAppMovieContext.cs`: Database context file (<xref:Microsoft.EntityFrameworkCore.DbContext>).
+
+The component files in the `MoviePages` folder are described in greater detail in the next part of this tutorial. The database context is described later in this article.
+
+ASP.NET Core is built with dependency injection, which is a software design pattern for achieving [Inversion of Control (IoC)](/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#dependency-inversion) between classes and their dependencies. Services, such as the EF Core database context, are registered with dependency injection during application startup. These services are injected into Razor components for use by the components.
+
+The <xref:Microsoft.AspNetCore.Components.QuickGrid> component is a Razor component for efficiently displaying data in tabular form. The scaffolder places a <xref:Microsoft.AspNetCore.Components.QuickGrid> component in the `Index` component (`Components/Pages/Index.razor`) to display movie entities. Calling <xref:Microsoft.Extensions.DependencyInjection.EntityFrameworkAdapterServiceCollectionExtensions.AddQuickGridEntityFrameworkAdapter%2A> on the service collection adds an EF Core adapter for <xref:Microsoft.AspNetCore.Components.QuickGrid> to recognize EF Core-supplied <xref:System.Linq.IQueryable%601> instances and to resolve database queries asynchronously for efficiency.
+
+The following code is added to the `Program` file by the scaffolder:
+
+```csharp
+builder.Services.AddDbContext<BlazorWebAppMoviesContext>(options =>
+    options.UseSqlServer(
+        builder.Configuration.GetConnectionString("BlazorWebAppMoviesContext") ?? 
+        throw new InvalidOperationException(
+            "Connection string 'BlazorWebAppMoviesContext' not found.")));
+
+builder.Services.AddQuickGridEntityFrameworkAdapter();
```

Comment:
The Blazor CRUD scaffolder was just updated to use an `IDbContextFactory` per our guidance:

```suggestion
builder.Services.AddDbContextFactory<BlazorApp17Context>(options =>
    options.UseSqlServer(
        builder.Configuration.GetConnectionString("BlazorWebAppMoviesContext") ??
        throw new InvalidOperationException(
            "Connection string 'BlazorWebAppMoviesContext' not found.")));

builder.Services.AddQuickGridEntityFrameworkAdapter();
```